### PR TITLE
Support custom configuration 

### PR DESCRIPTION
**Pull request by hossainemruz**
This PR fixes:
https://github.com/kubedb/project/issues/187
https://github.com/kubedb/project/issues/193

Task List:
- [x] Add support for custom configuration
- [x] Add test
- [x] Add documentation
Doc ref: https://github.com/kubedb/cli/pull/244

### CODE REVIEW

**Comment by tamalsaha** [comment]
- [ ] yq  static or dynamic binary
- [ ]

File: hack/docker/elasticsearch/5.6.4/config-merger.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -x
+configDir="/elasticsearch/config"
```
Comment (tamalsaha) [inline hack/docker/elasticsearch/5.6.4/config-merger.sh]: `CONFIG_DIR`

File: hack/docker/elasticsearch/5.6.4/Dockerfile [hunk 2]
```diff
@@ -41,6 +41,8 @@ RUN ln -s /fsloader/run_fsloader.sh /etc/service/fsloader/run
 RUN mkdir /etc/service/elasticsearch
 RUN ln -s /run.sh /etc/service/elasticsearch/run
 
+COPY yq /yq
```
Comment (tamalsaha) [inline hack/docker/elasticsearch/5.6.4/Dockerfile]: `COPY yq /usr/bin/yq`

File: hack/docker/elasticsearch/6.3.0/config/elasticsearch.yml [hunk 3]
```diff
@@ -30,6 +30,9 @@ discovery:
     ping.unicast.hosts: ${DISCOVERY_SERVICE}
     minimum_master_nodes: ${NUMBER_OF_MASTERS}
 
+xpack.security.enabled: ${XPACK_SECURITY_ENABLED:false}
+# ref: https://github.com/floragunncom/search-guard-ssl/issues/43
```
Comment (tamalsaha) [inline hack/docker/elasticsearch/6.3.0/config/elasticsearch.yml]: Comment above.

File: pkg/es-util/es_client_v5.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
+package es_util
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/appscode/go/crypto/rand"
+	api "github.com/kubedb/apimachinery/apis/kubedb/v1alpha1"
+	esv5 "gopkg.in/olivere/elastic.v5"
+	"gopkg.in/yaml.v2"
```
Comment (tamalsaha) [inline pkg/es-util/es_client_v5.go]: Use https://github.com/ghodss/yaml

File: pkg/es-util/database_util.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	CreateIndex(count int) error
+	CountIndex() (int, error)
+	GetIndexNames() ([]string, error)
+	GetAllNodesInfo() ([]NodeInfo, error)
+	GetElasticsearchSummary(indexName string) (*api.ElasticsearchSummary, error)
+	Stop()
+}
+
+type NodeSetting struct {
+	Name   string `yaml:"name,omitempty"`
```
Comment (tamalsaha) [inline pkg/es-util/database_util.go]: `json: `

Use https://github.com/ghodss/yaml

File: pkg/es-util/es_client_v6.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+	// Get analyzer
+	analyzerData, err := c.client.IndexGetSettings(indexName).Do(context.Background())
+	if err != nil {
+		return &api.ElasticsearchSummary{}, err
+	}
+
+	dataByte, err := json.Marshal(analyzerData[indexName].Settings["index"])
+	if err != nil {
+		return &api.ElasticsearchSummary{}, err
```
Comment (tamalsaha) [inline pkg/es-util/es_client_v6.go]: `return nil`

File: hack/docker/elasticsearch/5.6.4/config-merger.sh (hunk 1 above)
Comment (hossainemruz) [reply hack/docker/elasticsearch/5.6.4/config-merger.sh]: fixed

File: hack/docker/elasticsearch/5.6.4/Dockerfile (hunk 2 above)
Comment (hossainemruz) [reply hack/docker/elasticsearch/5.6.4/Dockerfile]: fixed

File: pkg/es-util/database_util.go (hunk 5 above)
Comment (hossainemruz) [reply pkg/es-util/database_util.go]: fixed

File: pkg/es-util/es_client_v5.go (hunk 4 above)
Comment (hossainemruz) [reply pkg/es-util/es_client_v5.go]: fixed

File: pkg/es-util/es_client_v6.go (hunk 6 above)
Comment (hossainemruz) [reply pkg/es-util/es_client_v6.go]: fixed

**Pull request merged**